sw/device/lib/dif/dif_pwrmgr.h

```diff
@@ -0,0 +1,497 @@
+  /**
+   * Despite low power mode being enabled and executing a wait for interrupt
+   * (WFI) instruction, an interrupt arrived at just the right time to break the
+   * executing core out of WFI.
+   */
+  kDifPwrmgrWakeupTypeFallThrough,
+  /**
+   * Despite low power mode being enabled and executing a wait for interrupt
+   * (WFI) instruction, an active flash, life cycle, or OTP operation was
+   * in progress when the power controller attempted to initiate low power
+   * entry.
+   */
+  kDifPwrmgrWakeupTypeAbort,
+} dif_pwrmgr_wakeup_type_t;
+
+/**
+ * Wake-up reason and source requests since the last time recording started.
+ */
+typedef struct dif_pwrmgr_wakeup_reason {
+  /**
+   * Wake-up type.
+   */
+  dif_pwrmgr_wakeup_type_t wakeup_type;
+  /**
+   * Peripheral sources that requested wake-up since the last time recording has
+   * started.
+   */
+  uint32_t sources;
+} dif_pwrmgr_wakeup_reason_t;
+
+/**
+ * Get wake-up reason and source requests since the last time recording started.
+ *
+ * The power manager automatically starts recording wake-up requests when it
+ * begins a valid low power entry. Recording continues until it is explicitly
+ * disabled by calling `dif_pwrmgr_wakeup_request_recording_enabled_set`. Thus,
+ * it is possible to record wake-up requests for multiple peripheral sources.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_config Whether power manager is recording wake-up reasons.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_wakeup_reason_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_wakeup_reason_t *reason);
+
+/**
+ * Clear wake-up reason and requests recorded during last low power state.
+ *
+ * @param pwrmgr A power manager.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_wakeup_reason_clear(const dif_pwrmgr_t *pwrmgr);
+
+/**
+ * Power manager interrupts.
+ */
+typedef enum dif_pwrmgr_irq {
+  /**
+   * The device woke up from low power state.
+   *
+   * Note: This interrupt is not triggered during power-on reset.
+   */
+  kDifPwrmgrIrqWakeUp,
+} dif_pwrmgr_irq_t;
+
+/**
+ * Enable or disable an interrupt.
+ *
+ * @param pwrmgr A power manager.
+ * @param irq An interrupt.
+ * @param state New state of the interrupt.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_irq_enabled_set(dif_pwrmgr_t *pwrmgr,
+                                               dif_pwrmgr_irq_t irq,
+                                               dif_pwrmgr_toggle_t state);
+
+/**
+ * Check if there is a pending request for an interrupt.
+ *
+ * @param pwrmgr A power manager.
+ * @param irq An interrupt.
+ * @param state State of the interrupt.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_irq_get(dif_pwrmgr_t *pwrmgr,
+                                       dif_pwrmgr_irq_t irq, bool *state);
```

Comment:
`bool` -> `dif_pwrmgr_toggle_t`

Reply:
This is not super important but the reason why I prefer `bool` over `dif_pwrmgr_toggle_t` in this case is that the state of an IRQ is not something you can control and also to avoid confusion with enabling/disabling interrupts.

Reply:
I personally agree with @jon-flatley on this one, I think the type is more descriptive in this case.

Reply:
Oh, wait, I guess this is the function to get the corresponding `PWRMGR.INTR_STATE` bit? In this case maybe a better name for the function would be `dif_pwrmgr_irq_pending_get`. I misinterpreted this function as "irq enabled get".

Reply:
This has been resolved during new DIF naming guidelines.